pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go

```diff
@@ -1818,6 +1818,9 @@ func (rs *replicaState) createReplicaSendStream(
 	rs.sendStream = &replicaSendStream{
 		parent: rs,
 	}
+	rs.sendStream.mu.Lock()
```

Comment:
This seems to be a no-op. If someone can "see" this `sendStream` concurrently before this method returns, then the bug is there. Otherwise, the `sendStream` initialization "happens before" all the uses by other goroutines.

Reply:
Okay I think I see what you mean -- but that actually isn't the case because it is re-entrant in some sense. We start emptying the send queue via:

https://github.com/kvoli/cockroach/blob/389a4661d2da12e3700402bee03aa41f3c6d9be5/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go#L1841-L1843

See this function:

https://github.com/kvoli/cockroach/blob/389a4661d2da12e3700402bee03aa41f3c6d9be5/pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go#L2529-L2539

The contending goroutine is from the callback which gets registered and enacted by the send token watcher (a different goroutine).